Incompatibility with latest bs4.

bs4 v4.13 makes a breaking change that causes failures like this: 

```
======================================================================
ERROR: test_css_split (compressor.tests.test_parsers.BeautifulSoupParserTests)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/Users/carlton/Projects/Django/django-compressor/compressor/tests/test_parsers.py", line 116, in test_css_split
    split = self.css_node.split_contents()
  File "/Users/carlton/Projects/Django/django-compressor/compressor/css.py", line 20, in split_contents
    and elem_attribs["rel"].lower() == "stylesheet"
AttributeError: 'AttributeValueList' object has no attribute 'lower'

----------------------------------------------------------------------
```

Need to track down in [the bs4 Changelog](https://git.launchpad.net/beautifulsoup/tree/CHANGELOG#n39). 

Test dependencies are currently pinned to 4.12.3.

Did a bisect and arrived a the following commit.

```
6d529cf1147dce8158cc06f94877a711caa4c788 is the first bad commit
commit 6d529cf1147dce8158cc06f94877a711caa4c788 (HEAD, origin/4.13-attributedict)
Author: Leonard Richardson <leonardr@segfault.org>
Date:   Mon May 27 11:13:00 2024 -0400

    Fixed type checking errors.

 bs4/__init__.py            | 5 +++--
 bs4/_typing.py             | 5 +++--
 bs4/builder/__init__.py    | 2 +-
 bs4/builder/_htmlparser.py | 3 ++-
 bs4/builder/_lxml.py       | 3 ++-
 bs4/element.py             | 1 +
 6 files changed, 12 insertions(+), 7 deletions(-)
```

https://git.launchpad.net/beautifulsoup/commit/?id=6d529cf1147dce8158cc06f94877a711caa4c788

The `rel` attribute has become a list and the following change fixes most of the tests, except for one:

```patch
diff --git a/compressor/css.py b/compressor/css.py
index baaf07f..7852510 100644
--- a/compressor/css.py
+++ b/compressor/css.py
@@ -17,7 +17,7 @@ class CssCompressor(Compressor):
             if (
                 elem_name == "link"
                 and "rel" in elem_attribs
-                and elem_attribs["rel"].lower() == "stylesheet"
+                and "stylesheet" in elem_attribs["rel"]
             ):
                 basename = self.get_basename(elem_attribs["href"])
                 filename = self.get_filename(basename)
```

Transforming the list items to lowercase cause all kinds of new test failures. 

```patch
diff --git a/compressor/css.py b/compressor/css.py
index baaf07f..519b3bb 100644
--- a/compressor/css.py
+++ b/compressor/css.py
@@ -17,7 +17,7 @@ class CssCompressor(Compressor):
             if (
                 elem_name == "link"
                 and "rel" in elem_attribs
-                and elem_attribs["rel"].lower() == "stylesheet"
+                and "stylesheet" in [rel.lower() for rel in elem_attribs["rel"]]
             ):
                 basename = self.get_basename(elem_attribs["href"])
                 filename = self.get_filename(basename)
```

The following test case fails after applying the first patch.

```pytb
__________________________________________________________________________ TemplatetagTestCase.test_uppercase_rel ___________________________________________________________________________

self = <compressor.tests.test_templatetags.TemplatetagTestCase testMethod=test_uppercase_rel>

        def test_uppercase_rel(self):
            template = """{% load compress %}{% compress css %}
    <link rel="StyleSheet" href="{{ STATIC_URL }}css/one.css" type="text/css">
    <style type="text/css">p { border:5px solid green;}</style>
    <link rel="StyleSheet" href="{{ STATIC_URL }}css/two.css" type="text/css">
    {% endcompress %}"""
            out = css_tag("/static/CACHE/css/output.600674ea1d3d.css")
>           self.assertEqual(out, render(template, self.context))
E           AssertionError: '<lin[22 chars]f="/static/CACHE/css/output.600674ea1d3d.css" type="text/css">' != '<lin[22 chars]f="/static/CACHE/css/output.919e93a2fea0.css" type="text/css">'
E           - <link rel="stylesheet" href="/static/CACHE/css/output.600674ea1d3d.css" type="text/css">
E           ?                                                       ^^^^^^  ^^^^
E           + <link rel="stylesheet" href="/static/CACHE/css/output.919e93a2fea0.css" type="text/css">
E           ?                                                       ^^^^^^^^^  ^

compressor/tests/test_templatetags.py:79: AssertionError
```

--- a/compressor/parser/beautifulsoup.py
+++ b/compressor/parser/beautifulsoup.py
@@ -12,7 +12,7 @@ class BeautifulSoupParser(ParserBase):
 
             # Disable multi_valued_attributes
             # http://www.crummy.com/software/BeautifulSoup/bs4/doc/#multi-valued-attributes
-            self.soup = BeautifulSoup(self.content, "html.parser", multi_valued_attributes={})
+            self.soup = BeautifulSoup(self.content, "html.parser", multi_valued_attributes=None)
         except ImportError as err:
             raise ImproperlyConfigured("Error while importing BeautifulSoup: %s" % err)
 

--- a/compressor/parser/beautifulsoup.py
+++ b/compressor/parser/beautifulsoup.py
@@ -10,7 +10,9 @@ class BeautifulSoupParser(ParserBase):
         try:
             from bs4 import BeautifulSoup
 
-            self.soup = BeautifulSoup(self.content, "html.parser")
+            # Disable multi_valued_attributes
+            # http://www.crummy.com/software/BeautifulSoup/bs4/doc/#multi-valued-attributes
+            self.soup = BeautifulSoup(self.content, "html.parser", multi_valued_attributes={})
         except ImportError as err:
             raise ImproperlyConfigured("Error while importing BeautifulSoup: %s" % err)
 
@@ -21,13 +23,7 @@ class BeautifulSoupParser(ParserBase):
         return self.soup.find_all("script")
 
     def elem_attribs(self, elem):
-        attrs = dict(elem.attrs)
-        # hack around changed behaviour in bs4, it returns lists now instead of one string, see
-        # http://www.crummy.com/software/BeautifulSoup/bs4/doc/#multi-valued-attributes
-        for key, value in attrs.items():
-            if type(value) is list:
-                attrs[key] = " ".join(value)
-        return attrs
+        return elem.attrs
 
     def elem_content(self, elem):
         return elem.string

--- a/requirements/tests.txt
+++ b/requirements/tests.txt
@@ -1,5 +1,5 @@
 Jinja2==3.1.6
-beautifulsoup4==4.12.3
+beautifulsoup4>=4.13
 brotli==1.1.0
 calmjs==3.4.4
 coverage==7.7.0

